prov/verbs/src/fi_verbs.c

```diff
@@ -390,6 +392,53 @@ static int fi_ibv_get_param_int(char *param_name, char *param_str,
 	return 0;
 }
 
+int fi_ibv_modify_qp(struct ibv_qp *qp)
+{
+	struct ibv_qp_attr attr = { 0 };
+	int ret;
+
+	attr.min_rnr_timer = verbs_min_rnr_timer;
+
+	ret = ibv_modify_qp(qp, &attr, IBV_QP_MIN_RNR_TIMER);
+	if (ret) {
+		VERBS_WARN(FI_LOG_EQ, "Unable to modify QP attribute\n");
+		return ret;
+	}
+	ret = fi_ibv_dbg_query_qp_attr(qp);
+	if (ret)
+		return ret;
+	return 0;
+}
+
+#if ENABLE_DEBUG
+int fi_ibv_dbg_query_qp_attr(struct ibv_qp *qp)
+{
+	struct ibv_qp_init_attr attr = { 0 };
+	struct ibv_qp_attr qp_attr = { 0 };
+	int ret;
+
+	ret = ibv_query_qp(qp, &qp_attr, IBV_QP_TIMEOUT | IBV_QP_RETRY_CNT |
+			   IBV_QP_RNR_RETRY | IBV_QP_MIN_RNR_TIMER, &attr);
+	if (ret) {
+		VERBS_WARN(FI_LOG_EP_CTRL, "Unable to query QP\n");
+		return ret;
+	}
+	FI_DBG_TRACE(&fi_ibv_prov, FI_LOG_EP_CTRL, "QP attributes: "
+		     "min_rnr_timer"	": %" PRIu8 ", "
+		     "timeout"		": %" PRIu8 ", "
+		     "retry_cnt"	": %" PRIu8 ", "
+		     "rnr_retry"	": %" PRIu8 "\n",
+		     qp_attr.min_rnr_timer, qp_attr.timeout, qp_attr.retry_cnt,
+		     qp_attr.rnr_retry);
+	return 0;
+}
+#else
+int fi_ibv_dbg_qp_attr(struct ibv_qp *qp)
+{
+	return 0;
+}
+#endif
```

Comment:
Name changed from debug to free build.  Please move up before use and make static.